doc/releases/changelog-dev.md

```diff
@@ -8,6 +8,9 @@
   Also added ``qml.pauli.trace_inner_product`` that can handle batches of dense matrices.
   [(#6811)](https://github.com/PennyLaneAI/pennylane/pull/6811)
 
+* Created a new ``qml.liealg`` module for advanved Lie algebra functionality.
```

Comment:
I _think_ it would be good to leave out the "advanced". First, computing the Lie closure is one of the first things users will need to do. Second, `liealg` will contain all functionality that is clearly associated to Lie algebras, right? So users that think that they do "non-advanced" stuff should also be advised to go to `qml.liealg`. :)